storm-client/src/jvm/org/apache/storm/windowing/WatermarkCountEvictionPolicy.java

```diff
@@ -17,20 +17,28 @@
  */
 package org.apache.storm.windowing;
 
+import org.apache.storm.streams.Pair;
+
+import java.util.concurrent.atomic.AtomicLong;
+
 /**
  * An eviction policy that tracks count based on watermark ts and
  * evicts events up to the watermark based on a threshold count.
  *
  * @param <T> the type of event tracked by this policy.
  */
-public class WatermarkCountEvictionPolicy<T> extends CountEvictionPolicy<T> {
+public class WatermarkCountEvictionPolicy<T> implements EvictionPolicy<T, Pair<Long, Long>> {
```

Comment:
Theres some code duplication, but the logic is not the same and delegation might not be correct here. Also it doesn't look very straightforward without a lot of refactoring.  Would like to keep it this way for now.